### ISSUE TITLE
Base64 password decoding

### ISSUE THREAD

**Issue by JeffO-KC** [bwp91/homebridge-ewelink#240]
### What issue do you have? Please be as thorough and explicit as possible.
I am US based and have installed a SONOFF iFan03 ceiling fan controller. Have installed eWeLink mobile app on iPhone. Configured iFan03 to be connected to wifi and can operate fan/light through eWeLink app. Installed homebridge-ewelink 6.2.0 and upgraded node to 14.16.1 as requested. I have uninstalled and reinstall homebridge-ewelink a couple times now. Happy to do it again if needed. I use the base64 username and enabled logging. All I see in the HOOBS log screen is (full log below):

4/17/2021, 12:48:37 PM [eWeLink] [v6.2.0] initialised. Syncing with eWeLink.
4/17/2021, 12:48:38 PM [eWeLink] Sending HTTP login() request.
4/17/2021, 12:48:40 PM [eWeLink] ***** Disabling plugin [v6.2.0]. *****

The iFan03 device never shows up in HOOBS. Not sure where to go from here since very little information is logged. I manually added the fanDevice but still does not show up.

{
            "platform": "eWeLink",
            "plugin_map": {
                "plugin_name": "homebridge-ewelink"
            },
            "name": "eWeLink",
            "mode": "auto",
            "username": "j*****y@yahoo.com",
            "password": "*****",
            "encodedPassword": true,
            "debug": true,
            "fanDevices": [
                {
                    "deviceId": "1000e8d29d",
                    "label": "Backporch Fan",
                    "hideLight": false,
                    "overrideDisabledLogging": true
                }
            ],
            "debugFakegato": true,
            "singleDevices": [],
            "multiDevices": [],
            "outletDevices": [],
            "lightDevices": [],
            "thDevices": [],
            "sensorDevices": [],
            "bridgeSensors": [],
            "groups": [],
            "ignoredDevices": []
        }

### Details of your setup.
* Do you use (1) Homebridge (+ config-ui), (2) Homebridge (CLI) or (3) HOOBS? 

HOOBS

* Which version of Homebridge/HOOBS do you have?

Bridge Name - HOOBS
HOOBS Version - 3.3.5
Node Version - 14.16.1
NPM - 7.9.0

* Which platform do you run Homebridge/HOOBS on (e.g. Raspberry Pi/Windows/HOOBS Box)? Please also mention your version of Node.js/NPM if known.

 Raspberry Pi

* Which version of this plugin (homebridge-ewelink) do you have? Has the issue started since upgrading from a previous version?

6.2.0 - No previous installs of this module. First time config. Although I have been updating as fast as you are releasing in hopes the issue would resolve.

* Which eWeLink devices do you have that are causing issues? Please include product models if applicable.

SONOFF iFan03

### Please paste any relevant logs below.

```
4/17/2021, 1:11:18 PM Initializing HAP-NodeJS v0.9.3...
4/17/2021, 1:11:21 PM Loaded plugin 'homebridge-apple-tv-remote'
4/17/2021, 1:11:22 PM [4/17/2021, 12:11:21 PM] Registering platform 'homebridge-apple-tv-remote.AppleTvPlatform'
4/17/2021, 1:11:28 PM Loaded plugin 'homebridge-broadlink-rm-pro'
4/17/2021, 1:11:29 PM [4/17/2021, 12:11:28 PM] Registering platform 'homebridge-broadlink-rm-pro.BroadlinkRM'
4/17/2021, 1:11:29 PM [4/17/2021, 12:11:29 PM] Registering platform 'homebridge-ewelink.eWeLink'
4/17/2021, 1:11:29 PM Loaded plugin 'homebridge-ewelink'
4/17/2021, 1:11:29 PM Loaded plugin 'homebridge-platform-wemo'
4/17/2021, 1:11:29 PM [4/17/2021, 12:11:29 PM] Registering platform 'homebridge-platform-wemo.BelkinWeMo'
4/17/2021, 1:11:32 PM Loaded plugin 'homebridge-ring'
4/17/2021, 1:11:32 PM [4/17/2021, 12:11:32 PM] Registering platform 'homebridge-ring.Ring'
4/17/2021, 1:11:32 PM Loaded plugin 'homebridge-samsung-tizen'
4/17/2021, 1:11:32 PM [4/17/2021, 12:11:32 PM] Registering platform 'homebridge-samsung-tizen.SamsungTizen'
4/17/2021, 1:11:32 PM Loaded plugin 'homebridge-shelly'
4/17/2021, 1:11:33 PM [4/17/2021, 12:11:33 PM] Registering platform 'homebridge-shelly.Shelly'
4/17/2021, 1:11:33 PM Loading 7 platforms...
4/17/2021, 1:11:33 PM [Wemo] Initializing BelkinWeMo platform...
4/17/2021, 1:11:33 PM [Ring] Initializing Ring platform...
4/17/2021, 1:11:33 PM [Shelly] Initializing Shelly platform...
4/17/2021, 1:11:33 PM [Samsung] Initializing SamsungTizen platform...
4/17/2021, 1:11:33 PM [Apple TV Platform] Initializing AppleTvPlatform platform...
4/17/2021, 1:11:33 PM [BroadlinkRM] Initializing BroadlinkRM platform...
4/17/2021, 1:11:33 PM [BroadlinkRM] [INFO] Automatically discovering Broadlink RM devices.
4/17/2021, 1:11:33 PM [BroadlinkRM] [INFO] Running Homebridge Broadlink RM Plugin version 4.4.2
4/17/2021, 1:11:33 PM [eWeLink] Initializing eWeLink platform...
4/17/2021, 1:11:33 PM [Ring] Configuring cached accessory  Front Door
4/17/2021, 1:11:33 PM [Ring] Configuring cached accessory  Hallway
4/17/2021, 1:11:33 PM [Wemo] ***** Disabling plugin [v3.0.5]. *****
4/17/2021, 1:11:33 PM [Wemo] ***** To change this, set disablePlugin to false. *****
4/17/2021, 1:11:33 PM [Shelly] 0 accessories loaded from cache
4/17/2021, 1:11:33 PM [Apple TV Platform] Initialing platform...
4/17/2021, 1:11:33 PM [Apple TV Platform] [Basement AppleTV Remote] Initializing...
4/17/2021, 1:11:33 PM [Apple TV Platform] [Basement AppleTV Remote] Adding on/off switch
4/17/2021, 1:11:33 PM [eWeLink] [v6.2.0] initialised. Syncing with eWeLink.
4/17/2021, 1:11:34 PM [eWeLink] Sending HTTP login() request.
4/17/2021, 1:11:34 PM [BroadlinkRM] No accessories have been added to the "undefined" platform config.
4/17/2021, 1:11:34 PM Initializing platform accessory 'Learn'...
4/17/2021, 1:11:34 PM Initializing platform accessory 'Scan Frequency'...
4/17/2021, 1:11:34 PM Preparing Advertiser for 'HOOBS 9E7A' using bonjour-hap backend!
4/17/2021, 1:11:34 PM Starting to advertise 'HOOBS 9E7A' using bonjour-hap backend!
4/17/2021, 1:11:34 PM Bridge is running on port 51826.
4/17/2021, 1:11:34 PM [Shelly] Admin server is running on port 8181
4/17/2021, 1:11:34 PM Preparing Advertiser for 'Basement TV EDF4' using bonjour-hap backend!
4/17/2021, 1:11:34 PM Starting to advertise 'Basement TV EDF4' using bonjour-hap backend!
4/17/2021, 1:11:34 PM Basement TV EDF4 is running on port 37573.
4/17/2021, 1:11:34 PM Please add [Basement TV EDF4] manually in Home app. Setup Code: 
4/17/2021, 1:11:36 PM [eWeLink] New HTTP API host received [us-apia.coolkit.cc].
4/17/2021, 1:11:36 PM [eWeLink] Sending HTTP login() request.
4/17/2021, 1:11:38 PM [BroadlinkRM] Learn getSwitchState: false
4/17/2021, 1:11:38 PM [BroadlinkRM] Scan Frequency getSwitchState: false
4/17/2021, 1:11:39 PM [eWeLink] ***** Disabling plugin [v6.2.0]. *****
4/17/2021, 1:11:40 PM [Ring] Found the following locations:
4/17/2021, 1:11:40 PM [Ring]   locationId: 
4/17/2021, 1:11:40 PM [Ring] Configuring 1 cameras and 2 devices for location 
4/17/2021, 1:11:41 PM [Apple TV Platform] [Basement AppleTV Remote] Connected
4/17/2021, 1:11:51 PM [BroadlinkRM] Learn getSwitchState: false
4/17/2021, 1:11:51 PM [BroadlinkRM] Scan Frequency getSwitchState: false
4/17/2021, 1:12:03 PM [BroadlinkRM] Learn getSwitchState: false
4/17/2021, 1:12:03 PM [BroadlinkRM] Scan Frequency getSwitchState: false
```

**Comment by bwp91** [comment]
Hi @JeffO-KC 
For some reason the plugin is being disabled, and for some even stranger reason the plugin isn't saying why.
Could you try using your plain text password (and turning off the `encodedPassword` option) to see if this has anything to do with why the plugin isnt working?
Thanks

**Comment by deon-wentzel** [comment]
Hi

I have six SONOFF iFAN03 running successfully. I am using HOOBS 3.3.5 with Node 14.15.1.  I use the same login credentials as I use on the EWE Link App.

I specify a config for each of the Fans under Fan Devices (which you have).

{
        "deviceId": "xxxxxxxxxxx",
        "hideLight": true,
        "label": "Spare Room Fan"
 } 

Every time I use the Fan via the Home APP, remote or the EWE Link App, I see a entry on the Logs, 

Apr 18 04:05:42 hoobs hoobs[639]: [eWeLink] [Devicexxxxxxx] current state [on].
Apr 18 04:05:42 hoobs hoobs[639]: [eWeLink] [Devicexxxxxxx] current speed [66%].

What I do not see in your config is a country code?

**Comment by deon-wentzel** [comment]
And my password is in pain text.

**Comment by JeffO-KC** [comment]
Boom! You pointed me in the right direction. It had to do with my password. I thought the special characters in it was causing issues so I changed it and apparently what I thought it was did not work. I reset my password and and changed the plugin to use plain test and everything started working. I then took my newly confirmed working password and used the following PowerShell to base64 it:

$Text = 'fancynewpassword'
$Bytes = [System.Text.Encoding]::Unicode.GetBytes($Text)
$EncodedText =[Convert]::ToBase64String($Bytes)
$EncodedText

I used the encoded text and set Encoded Password to yes and it reverted pack to having an issue with logging in and disabled the plugin again on re-init. I reverted back to plain text and it logged back in with no issue and I can see the iFan (fan and light) in HOOBS and in Apple Home.

I am good with using plain text. I try to default to some level of security/obscurity on these things but can live with it. No action needed for me to be happy but am more than willing to test if something needs testing.

**Comment by bwp91** [comment]
I have just released an update to the plugin which will trim spaces and new lines off passwords that have been encoded with base64. Different terminals, encoding websites, etc often put a new line or a space on the end of the string before encoding it.

You may find that your code is actually encoding
'fancynewpassword ' or 'fancynewpassword
'

Update, try your encoding method again to see if the update has helped.

**Comment by JeffO-KC** [comment]
A little more digging... this one is on me. Apologies! I told powershell to use unicode and it needed to be UTF8 encoded. I verified no extra spaces or crlf at the end and entered it in the plugin. It connected up just fine. If you need a powershell reference for base64 that works for the western world here is what did it (note the UTF8):

$Text = 'fancynewpassword'
$Bytes = [System.Text.Encoding]::UTF8.GetBytes($Text)
$EncodedText =[Convert]::ToBase64String($Bytes)
$EncodedText

Thanks for the prompt help. Again, apologies for the wasted time.

Cheers,
Jeff

**Comment by bwp91** [comment]
@JeffO-KC @deon-wentzel or anyone else with an iFan…

Could someone possibly install the current beta version of the plugin just to check all is working? Nothing should seem different but i’ve changed the commands slightly so just would like to get some feedback from someone with this device since i don’t own it myself

thanks!

**Comment by JeffO-KC** [comment]
Absolutely! 

sudo npm install homebridge-ewelink@beta -g

### OUTPUT

hoobs@hoobs:/home/hoobs/.hoobs/etc $ sudo npm install homebridge-ewelink@beta -g

changed 40 packages, and audited 41 packages in 27s

25 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
npm notice 
npm notice New minor version of npm available! 7.9.0 -> 7.11.1
npm notice Changelog: https://github.com/npm/cli/releases/tag/v7.11.1
npm notice Run npm install -g npm@7.11.1 to update!
npm notice 
hoobs@hoobs:/home/hoobs/.hoobs/etc $

### HOOBS Service restarted
Clean restart

### Log output shortened to just eWeLink entries...

4/24/2021, 2:09:40 PM Got SIGINT, shutting down Bridge...

4/24/2021, 2:09:49 PM Initializing HAP-NodeJS v0.9.3...
4/24/2021, 2:09:52 PM Loaded plugin 'homebridge-apple-tv-remote'
4/24/2021, 2:09:53 PM [4/24/2021, 1:09:52 PM] Registering platform 'homebridge-apple-tv-remote.AppleTvPlatform'
4/24/2021, 2:10:00 PM Loaded plugin 'homebridge-broadlink-rm-pro'
4/24/2021, 2:10:00 PM [4/24/2021, 1:10:00 PM] Registering platform 'homebridge-broadlink-rm-pro.BroadlinkRM'
4/24/2021, 2:10:00 PM Loaded plugin 'homebridge-ewelink'
4/24/2021, 2:10:00 PM [4/24/2021, 1:10:00 PM] Registering platform 'homebridge-ewelink.eWeLink'
4/24/2021, 2:10:00 PM Loaded plugin 'homebridge-platform-wemo'
4/24/2021, 2:10:00 PM [4/24/2021, 1:10:00 PM] Registering platform 'homebridge-platform-wemo.BelkinWeMo'
4/24/2021, 2:10:03 PM Loaded plugin 'homebridge-ring'
4/24/2021, 2:10:03 PM [4/24/2021, 1:10:03 PM] Registering platform 'homebridge-ring.Ring'
4/24/2021, 2:10:03 PM Loaded plugin 'homebridge-samsung-tizen'
4/24/2021, 2:10:03 PM [4/24/2021, 1:10:03 PM] Registering platform 'homebridge-samsung-tizen.SamsungTizen'
4/24/2021, 2:10:03 PM Loaded plugin 'homebridge-shelly'
4/24/2021, 2:10:04 PM [4/24/2021, 1:10:04 PM] Registering platform 'homebridge-shelly.Shelly'
4/24/2021, 2:10:04 PM Loading 7 platforms...

4/24/2021, 2:10:04 PM [eWeLink] Initializing eWeLink platform...

4/24/2021, 2:10:05 PM [eWeLink] [v6.2.1] initialised. Syncing with eWeLink.

4/24/2021, 2:10:05 PM Starting to advertise 'HOOBS 9E7A' using bonjour-hap backend!
4/24/2021, 2:10:05 PM Bridge is running on port 51826.

4/24/2021, 2:10:18 PM [eWeLink] LAN monitoring started.
4/24/2021, 2:10:19 PM [eWeLink] [Porch Fan] initialised and LAN mode unavailable as not supported.
4/24/2021, 2:10:19 PM [eWeLink] ✓ Setup complete. Thanks for using this plugin, I hope you find it helpful!

### Home app testing
Fan and light controls work
Siri voice commands work

### eWeLink app testing
Manual control of both fan and light via iOS app work

Beta testing looks good. No visible errors/issues and functionality works as did before beta install.

One thing I do notice and it is not due to the beta version is when "hideLight": false is set on the iFan the HOOBS ui does not show an icon for the light. It just looks kind of wonky. Here is a screen shot.

![image](https://user-images.githubusercontent.com/45737529/115970472-2391a100-a508-11eb-9a5a-7bde7065228f.png)


Let me know if you want this entered as its own issue.

Cheers,
Jeff

**Comment by bwp91** [comment]
@JeffO-KC I was wondering why your logs still said the old version of the plugin

```
4/24/2021, 2:10:05 PM [eWeLink] [v6.2.1] initialised. Syncing with eWeLink.
```

but then realised that you need to use a different command to install the beta on hoobs.

I would recommend reversing what you have done (installing the package globally, so this command uninstalls):

```
sudo npm uninstall homebridge-ewelink -g
```

And then please follow [these instructions ](https://github.com/bwp91/homebridge-ewelink/wiki/Beta-Version#hoobs-ui) to update to the beta on hoobs. Then when starting you should see `[v6.2.2-beta.5] initialised` in the logs.

Thanks for your willingness to try!

**Comment by JeffO-KC** [comment]
Sorry for the delay I was out of town for a bit.

### Install output
hoobs@hoobs:/home/hoobs/.hoobs $ cd ~/.hoobs/
hoobs@hoobs:/home/hoobs/.hoobs $ npm install homebridge-ewelink@beta

changed 1 package, and audited 611 packages in 17s

46 packages are looking for funding
  run `npm fund` for details

4 low severity vulnerabilities

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
hoobs@hoobs:/home/hoobs/.hoobs $

### Boot Logs
4/27/2021, 9:29:25 AM Initializing HAP-NodeJS v0.9.3...
4/27/2021, 9:29:28 AM Loaded plugin 'homebridge-apple-tv-remote'
4/27/2021, 9:29:29 AM [4/27/2021, 8:29:28 AM] Registering platform 'homebridge-apple-tv-remote.AppleTvPlatform'
4/27/2021, 9:29:35 AM Loaded plugin 'homebridge-broadlink-rm-pro'
4/27/2021, 9:29:35 AM [4/27/2021, 8:29:35 AM] Registering platform 'homebridge-broadlink-rm-pro.BroadlinkRM'
4/27/2021, 9:29:35 AM Loaded plugin 'homebridge-ewelink'
4/27/2021, 9:29:35 AM [4/27/2021, 8:29:35 AM] Registering platform 'homebridge-ewelink.eWeLink'
4/27/2021, 9:29:35 AM Loaded plugin 'homebridge-platform-wemo'
4/27/2021, 9:29:35 AM [4/27/2021, 8:29:35 AM] Registering platform 'homebridge-platform-wemo.BelkinWeMo'
4/27/2021, 9:29:39 AM Loaded plugin 'homebridge-ring'
4/27/2021, 9:29:39 AM [4/27/2021, 8:29:39 AM] Registering platform 'homebridge-ring.Ring'
4/27/2021, 9:29:39 AM Loaded plugin 'homebridge-samsung-tizen'
4/27/2021, 9:29:39 AM [4/27/2021, 8:29:39 AM] Registering platform 'homebridge-samsung-tizen.SamsungTizen'
4/27/2021, 9:29:39 AM Loaded plugin 'homebridge-shelly'
4/27/2021, 9:29:40 AM [4/27/2021, 8:29:40 AM] Registering platform 'homebridge-shelly.Shelly'
4/27/2021, 9:29:40 AM Loading 7 platforms...
4/27/2021, 9:29:40 AM [Wemo] Initializing BelkinWeMo platform...
4/27/2021, 9:29:40 AM [Ring] Initializing Ring platform...
4/27/2021, 9:29:40 AM [Shelly] Initializing Shelly platform...
4/27/2021, 9:29:40 AM [Samsung] Initializing SamsungTizen platform...
4/27/2021, 9:29:40 AM [Apple TV Platform] Initializing AppleTvPlatform platform...
4/27/2021, 9:29:40 AM [BroadlinkRM] Initializing BroadlinkRM platform...

4/27/2021, 9:29:40 AM [eWeLink] Initializing eWeLink platform...
4/27/2021, 9:29:40 AM [Ring] Configuring cached accessory Front Door
4/27/2021, 9:29:40 AM [Ring] Configuring cached accessory Hallway
4/27/2021, 9:29:40 AM [Wemo] ***** Disabling plugin [v3.0.5]. *****
4/27/2021, 9:29:40 AM [Wemo] ***** To change this, set disablePlugin to false. *****
4/27/2021, 9:29:40 AM [Shelly] 0 accessories loaded from cache
4/27/2021, 9:29:40 AM [Apple TV Platform] Initialing platform...

4/27/2021, 9:29:40 AM [eWeLink] [v6.2.2-beta.8] initialised. Syncing with eWeLink.
4/27/2021, 9:29:41 AM Preparing Advertiser for 'Stereo D6C9' using bonjour-hap backend!
4/27/2021, 9:29:41 AM Initializing platform accessory 'Learn'...
4/27/2021, 9:29:41 AM Initializing platform accessory 'Scan Frequency'...
4/27/2021, 9:29:41 AM Preparing Advertiser for 'HOOBS 9E7A' using bonjour-hap backend!
4/27/2021, 9:29:41 AM Starting to advertise 'Stereo D6C9' using bonjour-hap backend!
4/27/2021, 9:29:41 AM Stereo D6C9 is running on port 38739.
4/27/2021, 9:29:41 AM Starting to advertise 'HOOBS 9E7A' using bonjour-hap backend!
4/27/2021, 9:29:41 AM Bridge is running on port 51826.
4/27/2021, 9:29:41 AM [BroadlinkRM] Stereo getSwitchState: 0
4/27/2021, 9:29:41 AM Preparing Advertiser for 'Basement TV EDF4' using bonjour-hap backend!
4/27/2021, 9:29:41 AM Starting to advertise 'Basement TV EDF4' using bonjour-hap backend!
4/27/2021, 9:29:41 AM Basement TV EDF4 is running on port 35831.
4/27/2021, 9:29:41 AM Please add [Basement TV EDF4] manually in Home app.
4/27/2021, 9:29:46 AM [BroadlinkRM] Learn getSwitchState: false
4/27/2021, 9:29:46 AM [BroadlinkRM] Scan Frequency getSwitchState: false
4/27/2021, 9:29:46 AM [BroadlinkRM] Stereo getSwitchState: 0
4/27/2021, 9:29:48 AM [Ring] Found the following locations:
4/27/2021, 9:29:48 AM [Ring]   locationId: 9yuz1u-18mpn-0 - 
4/27/2021, 9:29:49 AM [Ring] Configuring 1 cameras and 2 devices for location 
4/27/2021, 9:29:52 AM [Apple TV Platform] [Basement AppleTV Remote] Connected
4/27/2021, 9:30:36 AM [BroadlinkRM] Stereo getSwitchState: 0
4/27/2021, 9:30:36 AM [BroadlinkRM] Learn getSwitchState: false
4/27/2021, 9:30:36 AM [BroadlinkRM] Scan Frequency getSwitchState: false

### Testing

I lost control of iFan from HOOBS and Home App.

eWeLink app still works

### Output from npm audit fix --force
hoobs@hoobs:/home/hoobs/.hoobs/etc $ npm audit fix --force
npm WARN using --force Recommended protections disabled.
npm WARN audit Updating homebridge-ring to 4.2.3,which is a SemVer major change.
npm WARN deprecated debug@4.1.1: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
npm WARN deprecated debug@4.1.1: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
npm WARN deprecated debug@4.1.1: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
npm WARN deprecated axios@0.19.2: Critical security vulnerability fixed in v0.21.1. For more information, see https://github.com/axios/axios/pull/3410

added 13 packages, removed 138 packages, changed 4 packages, and audited 487 packages in 30s

40 packages are looking for funding
  run `npm fund` for details

 npm audit report

axios  <0.21.1
Severity: high
Server-Side Request Forgery - https://npmjs.com/advisories/1594
fix available via `npm audit fix --force`
Will install homebridge-ring@9.17.0, which is a breaking change
node_modules/homebridge-ring/node_modules/axios
  homebridge-ring  4.0.0 - 9.0.0-beta.2
  Depends on vulnerable versions of axios
  node_modules/homebridge-ring

2 high severity vulnerabilities

To address all issues (including breaking changes), run:
  npm audit fix --force
hoobs@hoobs:/home/hoobs/.hoobs/etc $

### Restarted HOOBS services
eWeLink appears to initialize (log below)
Home app is working

### Boot Logs (just eWeLink entries)
4/27/2021, 9:41:10 AM Initializing HAP-NodeJS v0.9.3...
4/27/2021, 9:41:14 AM Loaded plugin 'homebridge-apple-tv-remote'
4/27/2021, 9:41:14 AM [4/27/2021, 8:41:14 AM] Registering platform 'homebridge-apple-tv-remote.AppleTvPlatform'
4/27/2021, 9:41:21 AM Loaded plugin 'homebridge-broadlink-rm-pro'
4/27/2021, 9:41:21 AM [4/27/2021, 8:41:21 AM] Registering platform 'homebridge-broadlink-rm-pro.BroadlinkRM'
4/27/2021, 9:41:21 AM Loaded plugin 'homebridge-ewelink'
4/27/2021, 9:41:21 AM [4/27/2021, 8:41:21 AM] Registering platform 'homebridge-ewelink.eWeLink'
4/27/2021, 9:41:21 AM Loaded plugin 'homebridge-platform-wemo'
4/27/2021, 9:41:21 AM [4/27/2021, 8:41:21 AM] Registering platform 'homebridge-platform-wemo.BelkinWeMo'
4/27/2021, 9:41:23 AM Loaded plugin 'homebridge-ring'
4/27/2021, 9:41:23 AM [4/27/2021, 8:41:23 AM] Registering platform 'homebridge-ring.Ring'
4/27/2021, 9:41:23 AM Loaded plugin 'homebridge-samsung-tizen'
4/27/2021, 9:41:23 AM [4/27/2021, 8:41:23 AM] Registering platform 'homebridge-samsung-tizen.SamsungTizen'
4/27/2021, 9:41:23 AM Loaded plugin 'homebridge-shelly'
4/27/2021, 9:41:24 AM [4/27/2021, 8:41:24 AM] Registering platform 'homebridge-shelly.Shelly'
4/27/2021, 9:41:24 AM Loading 7 platforms...

4/27/2021, 9:41:24 AM [eWeLink] Initializing eWeLink platform...

4/27/2021, 9:41:24 AM [eWeLink] [v6.2.2-beta.8] initialised. Syncing with eWeLink.

4/27/2021, 9:41:37 AM [eWeLink] LAN monitoring started.
4/27/2021, 9:41:37 AM [eWeLink] [Porch Fan] initialised and LAN mode unavailable as not supported.
4/27/2021, 9:41:37 AM [eWeLink] ✓ Setup complete. Thanks for using this plugin, I hope you find it helpful!

### Comments
Not sure if it was the npm audit or the simple fact I restarted HOOBS services that caused it to start working.

I notice the fan is not beeping when it changes state now... I and it is GREAT! Not sure if this was your doing or not but I LOVE it.

Happy to help!
Cheers ~Jeff
